dot/core/messages_integration_test.go

```diff
@@ -0,0 +1,172 @@
+	}
+
+	// Sign the transaction using Alice's key
+	err = ext.Sign(signature.TestKeyringPairAlice, o)
+	require.NoError(t, err)
+
+	extEnc, err := ctypes.EncodeToHexString(ext)
+	require.NoError(t, err)
+
+	extBytes := types.Extrinsic(common.MustHexToBytes(extEnc))
+	return extBytes
+}
+
+func TestService_HandleBlockProduced(t *testing.T) {
+	net := new(mocks.Network)
+	cfg := &Config{
+		Network:  net,
+		Keystore: keystore.NewGlobalKeystore(),
+	}
+
+	s := NewTestService(t, cfg)
+	err := s.Start()
+	require.Nil(t, err)
+
+	// simulate block sent from BABE session
+	digest := types.NewDigest()
+	prd, err := types.NewBabeSecondaryPlainPreDigest(0, 1).ToPreRuntimeDigest()
+	require.NoError(t, err)
+	err = digest.Add(*prd)
+	require.NoError(t, err)
+
+	newBlock := types.Block{
+		Header: types.Header{
+			Number:     big.NewInt(1),
+			ParentHash: s.blockState.BestBlockHash(),
+			Digest:     digest,
+		},
+		Body: *types.NewBody([]types.Extrinsic{}),
+	}
+
+	expected := &network.BlockAnnounceMessage{
+		ParentHash:     newBlock.Header.ParentHash,
+		Number:         newBlock.Header.Number,
+		StateRoot:      newBlock.Header.StateRoot,
+		ExtrinsicsRoot: newBlock.Header.ExtrinsicsRoot,
+		Digest:         digest,
+		BestBlock:      true,
+	}
+
+	net.On("GossipMessage", expected)
+
+	state, err := s.storageState.TrieState(nil)
+	require.NoError(t, err)
+
+	err = s.HandleBlockProduced(&newBlock, state)
+	require.NoError(t, err)
+
+	time.Sleep(time.Second)
+	net.AssertCalled(t, "GossipMessage", expected)
+}
+
+func TestService_HandleTransactionMessage(t *testing.T) {
+	t.Parallel()
+
+	const peer1 = "testPeer1"
+
+	kp, err := sr25519.GenerateKeypair()
+	require.NoError(t, err)
+
+	ks := keystore.NewGlobalKeystore()
+	ks.Acco.Insert(kp)
+
+	ctrl := gomock.NewController(t)
+	telemetryMock := NewMockClient(ctrl)
+	telemetryMock.EXPECT().SendMessage(gomock.Any()).AnyTimes()
+
+	cfg := &Config{
+		Keystore:         ks,
+		TransactionState: state.NewTransactionState(telemetryMock),
+	}
+
+	s := NewTestService(t, cfg)
+	genHash := s.blockState.GenesisHash()
+	genHeader, err := s.blockState.BestBlockHeader()
+	require.NoError(t, err)
+
+	rt, err := s.blockState.GetRuntime(nil)
+	require.NoError(t, err)
+
+	ts, err := s.storageState.TrieState(nil)
+	require.NoError(t, err)
+	rt.SetContextStorage(ts)
+
+	block := sync.BuildBlock(t, rt, genHeader, nil)
+
+	err = s.handleBlock(block, ts)
+	require.NoError(t, err)
+
+	extBytes := createExtrinsic(t, rt, genHash, 0)
+	msg := &network.TransactionMessage{Extrinsics: []types.Extrinsic{extBytes}}
+	b, err := s.HandleTransactionMessage(peer1, msg)
+	require.NoError(t, err)
+	require.True(t, b)
+
+	pending := s.transactionState.(*state.TransactionState).Pending()
+	require.NotEqual(t, 0, len(pending))
+	require.Equal(t, extBytes, pending[0].Extrinsic)
+
+	extBytes = []byte(`bogus extrinsic`)
+	msg = &network.TransactionMessage{Extrinsics: []types.Extrinsic{extBytes}}
+	b, err = s.HandleTransactionMessage(peer1, msg)
```

Comment:
What's `b` here?

Reply:
I am not sure, i just moved existing tests over I didn't write the integration tests in this PR

Reply:
its a boolean to indicate if the txns should be propagated. Renamed to shouldPropagate